src/envoy/tcp/tcp_cluster_rewrite/config.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "envoy/config/filter/network/tcp_cluster_rewrite/v2alpha1/config.pb.h"
+
+#include "envoy/network/connection.h"
+#include "envoy/network/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+using namespace ::istio::envoy::config::filter::network::tcp_cluster_rewrite;
+
+namespace Envoy {
+namespace Tcp {
+namespace TcpClusterRewrite {
+
+/**
+ * Config registration for the TCP cluster rewrite filter. @see
+ * NamedNetworkFilterConfigFactory.
+ */
+class TcpClusterRewriteFilterConfigFactory
```

Comment:
use factory base

Reply:
That causes a compile error because we don't have `*.pb.validate.h` files here.

https://github.com/envoyproxy/envoy/blob/3e15c94901f7ce5ebca671e451efceec8053da86/source/extensions/filters/network/common/factory_base.h#L29